alibi/explainers/similarity/backends/pytorch/base.py

```diff
@@ -0,0 +1,97 @@
+"""Torch backend for similarity explainers.
+
+Methods unique to the Torch backend are defined here. The interface this class defines syncs with the torch backend in
+order to ensure that the similarity methods only require to match this interface.
+"""
+
+import random
+from typing import Callable
+
+import numpy as np
+import torch.nn as nn
+import torch
+
+
+class _TorchBackend(object):
+    device = None
+
+    @staticmethod
+    def get_grads(
+            model: nn.Module,
+            x: torch.Tensor,
+            y: torch.Tensor,
+            loss_fn: Callable[[torch.Tensor, torch.Tensor], torch.Tensor],
+    ) -> np.ndarray:
+        """
+        Computes the gradients of the loss function with respect to the model's parameters for a single training and
+        target pair.
+
+        Parameters:
+        -----------
+        model:
+            The model to compute gradients for.
+        x:
+            The input data.
+        y:
+            The target data.
+        loss_fn:
+            The loss function to use.
+
+        Returns:
+        --------
+        grads:
+            The gradients of the loss function with respect to the model's parameters. This is returned as a flattened \
+            array.
+        """
+
+        for param in model.parameters():
+            if isinstance(param.grad, torch.Tensor):
+                param.grad.data.zero_()
```

Comment:
Why not using `model.zero_grad()` ? 
If not possible, can you remind me why the  `if`  statement necessary? 
Is it because you can have `None` when for example a layer does not have a bias?
Does it cover every possible scenario?

Reply:
No reason, I'm not sure why I did it this way. Will change it to `model.zero_grad()`. 

As I recall the if statement is needed because `param.grad` is `None` prior to the backward pass.